src/entities/trade.ts

```diff
@@ -174,10 +194,13 @@ export class Trade<TInput extends Currency, TOutput extends Currency, TTradeType
     let spotOutputAmount = CurrencyAmount.fromRawAmount(this.outputAmount.currency, 0)
     for (const { route, inputAmount } of this.swaps) {
       const midPrice = route.midPrice
-      spotOutputAmount = spotOutputAmount.add(midPrice.quote(inputAmount))
+
+      const postTaxInputAmount = inputAmount.multiply(new Fraction(ONE).subtract(this.inputTax))
+      spotOutputAmount = spotOutputAmount.add(midPrice.quote(postTaxInputAmount))
     }
 
-    const priceImpact = spotOutputAmount.subtract(this.outputAmount).divide(spotOutputAmount)
+    const preTaxOutputAmount = this.outputAmount.divide(new Fraction(ONE).subtract(this.outputTax))
+    const priceImpact = spotOutputAmount.subtract(preTaxOutputAmount).divide(spotOutputAmount)
```

Comment:
nit - do we need to distinguish the math between exact in and exact out for FOT swap?